File: fidessa/Triangle.java
import java.util.Scanner;

public class Triangle {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int lines = in.nextInt();
        for (int i = 0; i < lines; i++) {
            int a = in.nextInt();
            int b = in.nextInt();
            int c = in.nextInt();
            if (a < 0 || b < 0 || c < 0) {
                System.out.println("None of these");
                continue;
            }
            if (a + b <= c || b + c <= a || a + c <= b) {
                System.out.println("None of these");
            } else if (a == b && b == c) {
                System.out.println("Equilateral");
            } else if (a == b || b == c || a ==c) {
                System.out.println("Isosceles");
            } else {
                System.out.println("None of these");
            }
        }
    }
}

File: fidessa/Triangle.cpp
# include <iostream>
using namespace std;

int main() {
    int lines;
    cin >> lines;
    for (int i = 0; i < lines; i++) {
        int a, b, c;
        cin >> a >> b >> c;

        if (a <0 || b < 0 || c < 0) {
            cout << "None of these" << endl;
            continue;
        }

        if (a + b <= c || b + c <= a ||
            a + c <= b) {
            cout << "None of these" << endl;
            continue;
        } else if (a == b && b == c) {
            cout << "Equilateral" << endl;
        } else  if (a == b || b == c || a == c) {
            cout << "Isosceles" << endl;
        } else {
            cout << "None of these" << endl;
        }
    }
    return 0;
}
